test/yb_test.c

```diff
@@ -0,0 +1,339 @@
+ * in compliance with the License. You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <math.h>
+#include <time.h>
+#include <errno.h>
+#include <string.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/resource.h>
+#include <unistd.h>
+
+#include <hdr/hdr_histogram.h>
+#include <hdr/hdr_histogram_log.h>
+
+#include "minunit.h"
+
+#if defined(_MSC_VER)
+#pragma warning(push)
+#pragma warning(disable: 4996)
+#endif
+
+int tests_run = 0;
+int yb_default_max = 16777215;
+int yb_default_bucket_factor = 16;
+
+static int get_subbucket_width(hdr_histogram* h, int value)
+{
+    return hdr_next_non_equivalent_value(h, value) - lowest_equivalent_value(h, value);
+}
+
+static bool compare_values(double a, double b, double variation)
+{
+    return compare_double(a, b, b * variation);
+}
+
+static char* print_subbuckets(hdr_histogram* h)
+{
+    struct hdr_iter iter;
+    hdr_iter_init(&iter, h);
+    while(hdr_iter_next(&iter))
+    {
+        printf("index: %d, [%lld-%lld), count: %"COUNT_PRINT_FORMAT" \n",iter.counts_index, iter.value_iterated_to,
+            iter.highest_equivalent_value + 1, iter.count);
+    }
+    return 0;
+}
+
+/* 
+ * histogram initialization and counts len test
+ */
+static char* yb_test_create(void)
+{
+    /* 
+     * counts array not allocated here, would need calloc(sizeof(hdr_histogram) + 176 * 
+     * sizeof(count_t)) for full allocation
+     */
+    #ifdef FLEXIBLE_COUNTS_ARRAY
+    hdr_histogram* h = (hdr_histogram*) calloc(1, sizeof(hdr_histogram));
+    int r = yb_hdr_init(1, yb_default_max, yb_default_bucket_factor, h);
+    #else
+    hdr_histogram* h;
+    int r = hdr_init(1, yb_default_max, 1, &h);
+    #endif
+
+    mu_assert("Failed to allocate hdr_histogram", r == 0);
+    mu_assert("Failed to allocate hdr_histogram", h != NULL);
+
+    #ifdef FLEXIBLE_COUNTS_ARRAY
+    mu_assert("Incorrect counts array length", compare_int64(h->counts_len, 176));
+    #else
+    mu_assert("Incorrect counts array length", compare_int64(h->counts_len, 336));
+    #endif
+
+    hdr_close(h);
+
+    return 0;
+}
+
+/* 
+ * percentile calc test ported over from hdr_histogram_test.c
+ */
+static char* yb_test_create_with_large_values(void)
+{
+    #ifdef FLEXIBLE_COUNTS_ARRAY
+    hdr_histogram* dummy = (hdr_histogram*) calloc(1, sizeof(hdr_histogram));
+    yb_hdr_init(20000000, 100000000, 32, dummy);
+    hdr_histogram* h = (hdr_histogram*) calloc(1, sizeof(hdr_histogram) + 
+        dummy->counts_len * sizeof(count_t));
```

Comment:
It took me a while to realize you were using `dummy` to get the `counts_len` here. Could you give a comment to make this more clear?

Reply:
Actually, you use this pattern in almost every test. Might it make sense to extract it out as a separate function or macro?